ctapipe/image/extractor.py

```diff
@@ -124,11 +150,19 @@ def extract_pulse_time_weighted_average(waveforms):
         Shape: (n_chan, n_pix)
 
     """
-    samples_i = np.indices(waveforms.shape)[2]
-    pulse_time = np.average(samples_i, weights=waveforms, axis=2)
-    outside = np.logical_or(pulse_time < 0, pulse_time >= waveforms.shape[2])
-    pulse_time[outside] = -1
-    return pulse_time
+    n_samples = waveforms.size
+    start = peak_index - shift
+    end = start + width
+
+    num = 0
+    den = 0
+    for isample in prange(start, end):
+        if 0 <= isample < n_samples:
+            num += waveforms[isample] * isample
+            den += waveforms[isample]
+
+    # TODO: Return pulse time in units of ns instead of isample
+    ret[0] = num / den if den > 0 else peak_index
```

Comment:
I think you might be missing a `return` statement here

Reply:
that will also fix the errors in Codacy (since it complains about this)

Reply:
No, this is how `@guvectorize` works, the return value is stored in the `ret` array

Reply:
Like filling a vector in c++ that was passed to a function by reference

Reply:
ah, I see - then we can ignore the codacy errors